runner/conmon_test/conmon_test.go

```diff
@@ -101,58 +101,170 @@ var _ = Describe("conmon", func() {
+				conmon.WithPath(conmonPath),
+				conmon.WithContainerID(ctrID),
+				conmon.WithContainerUUID(ctrID),
+				conmon.WithRuntimePath(validPath),
+				conmon.withLogPath("k8s-file"),
+			)
+			Expect(stderr).To(ContainSubstring("k8s-file requires a filename"))
+		})
+		It("k8s-file: requires a filename", func() {
+			_, stderr := getConmonOutputGivenOptions(
+				conmon.WithPath(conmonPath),
+				conmon.WithContainerID(ctrID),
+				conmon.WithContainerUUID(ctrID),
+				conmon.WithRuntimePath(validPath),
+				conmon.withLogPath("k8s-file:"),
+			)
+			Expect(stderr).To(ContainSubstring("k8s-file requires a filename"))
+		})
 		It("log driver as path should pass", func() {
 			_, stderr := getConmonOutputGivenOptions(
 				conmon.WithPath(conmonPath),
 				conmon.WithContainerID(ctrID),
 				conmon.WithContainerUUID(ctrID),
 				conmon.WithRuntimePath(validPath),
 				conmon.WithLogDriver("", tmpLogPath),
+				conmon.WithLogLevel("debug"),
 			)
-			Expect(stderr).To(BeEmpty())
+			Expect(stderr).To(ContainSubstring("LogDriver K8S TRUE k8s_file "+tmpLogPath))
+			Expect(stderr).To(ContainSubstring("LogDriver JOURNALD FALSE"))
+
+			_, err := os.Stat(tmpLogPath)
+			Expect(err).To(BeNil())
+		})
+		It("log driver as k8s-file:path should pass", func() {
+			_, stderr := getConmonOutputGivenOptions(
+				conmon.WithPath(conmonPath),
+				conmon.WithContainerID(ctrID),
+				conmon.WithContainerUUID(ctrID),
+				conmon.WithRuntimePath(validPath),
+				conmon.WithLogDriver("k8s-file", tmpLogPath),
+				conmon.WithLogLevel("debug"),
+			)
+			Expect(stderr).To(ContainSubstring("LogDriver K8S TRUE k8s_file "+tmpLogPath))
+			Expect(stderr).To(ContainSubstring("LogDriver JOURNALD FALSE"))
+
+			_, err := os.Stat(tmpLogPath)
+			Expect(err).To(BeNil())
+		})
+		It("log driver as :path should pass", func() {
+			_, stderr := getConmonOutputGivenOptions(
+				conmon.WithPath(conmonPath),
+				conmon.WithContainerID(ctrID),
+				conmon.WithContainerUUID(ctrID),
+				conmon.WithRuntimePath(validPath),
+				conmon.withLogPath(":"+tmpLogPath),
+				conmon.WithLogLevel("debug"),
+			)
+			Expect(stderr).To(ContainSubstring("LogDriver K8S TRUE k8s_file "+tmpLogPath))
+			Expect(stderr).To(ContainSubstring("LogDriver JOURNALD FALSE"))
 
 			_, err := os.Stat(tmpLogPath)
 			Expect(err).To(BeNil())
 		})
 		It("log driver as none should pass", func() {
+			os.Remove("none")
+
 			_, stderr := getConmonOutputGivenOptions(
 				conmon.WithPath(conmonPath),
 				conmon.WithContainerID(ctrID),
 				conmon.WithContainerUUID(ctrID),
 				conmon.WithRuntimePath(validPath),
 				conmon.WithLogDriver("none", ""),
+				conmon.WithLogLevel("debug"),
 			)
-			Expect(stderr).To(BeEmpty())
+			Expect(stderr).To(ContainSubstring("LogDriver K8S FALSE k8s_file (null)"))
+			Expect(stderr).To(ContainSubstring("LogDriver JOURNALD FALSE"))
+
+			_, err := os.Stat("none")
+			Expect(err).NotTo(BeNil())
 		})
 		It("log driver as off should pass", func() {
+			os.Remove("off")
```

Comment:
this should be in a file in a tmpdir for idempotency

Reply:
also can we put "off" in a variable so we can change it in one place :)

Reply:
hmmm nevermind I see what's happening here....

Reply:
what happens if we just don't remove this file? I'm fine hitting a flake if this file happens to exist

Reply:
Well, now that the test passes, that's fine.  It'll just flake.  Before when the test was failing (i.e. conmon rc17), --log-path off writes to a file called off, and I'd have to manually rm the file as I was developing the code to fix that.

Since conmon shouldn't actually write those files anymore, I can get behind removing the rm

Reply:
Removed the os.Remove, and chdir to the tmpdir so any incidentally created files are contained (and the folder is automatically cleaned before/after each test)